dvc/remote/ssh/connection.py

```diff
@@ -110,7 +114,13 @@ def makedirs(self, path):
             self.makedirs(head)
 
         if tail:
-            self._sftp.mkdir(path)
+            try:
+                self._sftp.mkdir(path)
+            except IOError as e:
+                # Since paramiko errors are very vague we need to recheck
+                # whether it's because path already exists or something else
+                if e.errno == errno.EACCES or not self.exists(path):
```

Comment:
Is EEXIST not available?

Reply:
Only `ENOENT` and `EACCES`. `IOError` without errno is used otherwise. I may not check for code at all only for exists for clarity.